silhouette/app/com/mohiva/play/silhouette/impl/providers/state/CsrfStateItemHandler.scala

```diff
@@ -0,0 +1,167 @@
+package com.mohiva.play.silhouette.impl.providers.state
+
+import javax.inject.Inject
+
+import com.mohiva.play.silhouette.api.util.{ ExtractableRequest, IDGenerator }
+import com.mohiva.play.silhouette.impl.providers.SocialStateItem.ItemStructure
+import com.mohiva.play.silhouette.impl.providers.{ PublishableSocialStateItemHandler, SocialStateItem, SocialStateItemHandler }
+import play.api.libs.json.{ Format, JsValue, Json }
+import CsrfStateItemHandler._
+import com.mohiva.play.silhouette.api.crypto.CookieSigner
+import com.mohiva.play.silhouette.impl.exceptions.OAuth2StateException
+import play.api.mvc.{ Cookie, RequestHeader, Result }
+
+import scala.concurrent.duration._
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.{ Failure, Success, Try }
+
+case class CsrfState(value: String) extends SocialStateItem
+
+class CsrfStateItemHandler @Inject() (
+  settings: CsrfStateSettings,
+  idGenerator: IDGenerator,
+  cookieSigner: CookieSigner) extends SocialStateItemHandler with PublishableSocialStateItemHandler {
+
+  /**
+   * The item the handler can handle.
+   */
+  override type Item = CsrfState
+
+  /**
+   * Gets the state item the handler can handle.
+   *
+   * @param ec The execution context to handle the asynchronous operations.
+   * @return The state params the handler can handle.
+   */
+  override def item(implicit ec: ExecutionContext): Future[Item] = {
+    idGenerator.generate.map { CsrfState(_) }
+  }
+
+  /**
+   * Indicates if a handler can handle the given [[SocialStateItem]].
+   *
+   * This method should check if the [[serialize]] method of this handler can serialize the given
+   * unserialized state item.
+   *
+   * @param item The item to check for.
+   * @return [[Some]] casted state item if the handler can handle the given state item, [[None]] otherwise.
+   */
+  override def canHandle(item: SocialStateItem): Option[Item] = item match {
+    case i: Item => Some(i)
+    case _       => None
+  }
+
+  /**
+   * Indicates if a handler can handle the given unserialized state item.
+   *
+   * This method should check if the [[unserialize]] method of this handler can unserialize the given
+   * serialized state item.
+   *
+   * @param item The item to check for.
```

Comment:
@Saheb Sorry, I've not seen your comment.

The param description should be aligned in one line rather than with only using one space after the param name and the description.

The correct way would be to search the longest param name. In this case it's request. Then use one space and write the description. Then align all other descriptions, so that they are aligned in one vertical line.
```
* @param item    The item to publish.
* @param result  The result to send to the client.
* @param request The current request.
```

Instead of:
```
* @param item The item to publish.
* @param result The result to send to the client.
* @param request The current request.
```

A description starts also with an uppercase letter and ends with a dot. It's a real sentence. It's the same as with normal doc comments, they must end with a dot.